datalake/src/main/java/com/sequenceiq/datalake/flow/loadbalancer/dns/UpdateLoadBalancerDNSActions.java

```diff
@@ -0,0 +1,117 @@
+package com.sequenceiq.datalake.flow.loadbalancer.dns;
+
+import static com.sequenceiq.datalake.flow.loadbalancer.dns.UpdateLoadBalancerDNSEvent.UPDATE_LOAD_BALANCER_DNS_FAILURE_HANDLED_EVENT;
+import static com.sequenceiq.datalake.flow.loadbalancer.dns.UpdateLoadBalancerDNSEvent.UPDATE_LOAD_BALANCER_DNS_SUCCESS_EVENT;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.StackV4Endpoint;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGeneratorFactory;
+import com.sequenceiq.cloudbreak.event.ResourceEvent;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.events.EventSenderService;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.loadbalancer.dns.event.StartUpdateLoadBalancerDNSEvent;
+import com.sequenceiq.datalake.flow.loadbalancer.dns.event.UpdateLoadBalancerDNSFailedEvent;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.FlowEvent;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.flow.core.FlowState;
+
+@Configuration
+public class UpdateLoadBalancerDNSActions {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdateLoadBalancerDNSActions.class);
+
+    @Inject
+    private SdxService sdxService;
+
+    @Inject
+    private StackV4Endpoint stackV4Endpoint;
+
+    @Inject
+    private RegionAwareInternalCrnGeneratorFactory regionAwareInternalCrnGeneratorFactory;
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private EventSenderService eventSenderService;
+
+    @Bean(name = "UPDATE_LOAD_BALANCER_DNS_STATE")
+    public Action<?, ?> updateLoadBalancerDNSAction() {
+        return new AbstractSdxAction<>(StartUpdateLoadBalancerDNSEvent.class) {
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters,
+                    StateContext<FlowState, FlowEvent> stateContext,
+                    StartUpdateLoadBalancerDNSEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, StartUpdateLoadBalancerDNSEvent payload,
+                    Map<Object, Object> variables) {
+                SdxCluster sdxCluster = sdxService.getById(payload.getResourceId());
```

Comment:
Could you please move the logic from here to a service (e.g. UpdateLoadBalancerDNSService).